### PR TITLE
reloader: don't fail on envvar expansion errors

### PR DESCRIPTION
**Pull request by rexagod**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Refer: https://github.com/thanos-io/thanos/pull/7429
Fixes: https://github.com/prometheus-operator/prometheus-operator/issues/6136

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

```
/ $ wget -qO- localhost:8080/metrics | grep reloader_config
# HELP reloader_config_apply_operations_failed_total Total number of config apply operations that failed.
# TYPE reloader_config_apply_operations_failed_total counter
reloader_config_apply_operations_failed_total 0
# HELP reloader_config_apply_operations_total Total number of config apply operations.
# TYPE reloader_config_apply_operations_total counter
reloader_config_apply_operations_total 17
# HELP reloader_config_env_var_expansion_errors Number of environment variable expansions that failed during the last operation.
# TYPE reloader_config_env_var_expansion_errors gauge
reloader_config_env_var_expansion_errors 2
```

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
- Don't fail reloader on environment variable expansion errors when expanding configurations. Refer the upstream [changelog](https://github.com/thanos-io/thanos/pull/7429/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR28) for more details.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -143,5 +143,6 @@ require (
 replace (
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring => ./pkg/apis/monitoring
 	github.com/prometheus-operator/prometheus-operator/pkg/client => ./pkg/client
+	github.com/thanos-io/thanos => github.com/rexagod/thanos v0.0.0-20240618135941-2c88397c37c1
```
Comment (rexagod) [inline go.mod:146]: (will drop once upstream PR is in)

File: go.mod [hunk 2]
```diff
@@ -25,11 +25,11 @@ require (
 	github.com/prometheus-operator/prometheus-operator/pkg/client v0.74.0
 	github.com/prometheus/alertmanager v0.27.0
 	github.com/prometheus/client_golang v1.19.1
-	github.com/prometheus/common v0.54.0
+	github.com/prometheus/common v0.54.1-0.20240615204547-04635d2962f9
```
Comment (simonpasquier) [inline go.mod:28]: can you rebase on main? the version is more recent.

**Comment by rexagod** [comment]
I'll redo the links (https://github.com/prometheus-operator/website/pull/79).

**Review by simonpasquier — APPROVED** [review]
Thanks!

**Review by ArthurSens — APPROVED** [review]
Thanks for also fixing the links!

**Pull request merged**